datafusion/core/src/datasource/file_format/parquet.rs

```diff
@@ -256,167 +253,52 @@ impl FileFormat for ParquetFormat {
-                                min_values[i] = None;
-                            }
-                        }
-                    }
-                    return;
-                }
-            }
-            max_values[i] = None;
-            min_values[i] = None;
-        }
-        ParquetStatistics::Float(s) => {
-            if let DataType::Float32 = fields[i].data_type() {
-                if s.has_min_max_set() {
-                    if let Some(max_value) = &mut max_values[i] {
-                        match max_value.update_batch(&[Arc::new(Float32Array::from(
-                            vec![Some(*s.max())],
-                        ))]) {
-                            Ok(_) => {}
-                            Err(_) => {
-                                max_values[i] = None;
-                            }
-                        }
-                    }
-                    if let Some(min_value) = &mut min_values[i] {
-                        match min_value.update_batch(&[Arc::new(Float32Array::from(
-                            vec![Some(*s.min())],
-                        ))]) {
-                            Ok(_) => {}
-                            Err(_) => {
-                                min_values[i] = None;
-                            }
-                        }
-                    }
-                    return;
-                }
-            }
-            max_values[i] = None;
-            min_values[i] = None;
-        }
-        ParquetStatistics::Double(s) => {
-            if let DataType::Float64 = fields[i].data_type() {
-                if s.has_min_max_set() {
-                    if let Some(max_value) = &mut max_values[i] {
-                        match max_value.update_batch(&[Arc::new(Float64Array::from(
-                            vec![Some(*s.max())],
-                        ))]) {
-                            Ok(_) => {}
-                            Err(_) => {
-                                max_values[i] = None;
-                            }
-                        }
-                    }
-                    if let Some(min_value) = &mut min_values[i] {
-                        match min_value.update_batch(&[Arc::new(Float64Array::from(
-                            vec![Some(*s.min())],
-                        ))]) {
-                            Ok(_) => {}
-                            Err(_) => {
-                                min_values[i] = None;
-                            }
-                        }
-                    }
-                    return;
-                }
-            }
-            max_values[i] = None;
-            min_values[i] = None;
-        }
-        _ => {
-            max_values[i] = None;
-            min_values[i] = None;
+) -> ColumnStatistics {
+    let (min_value, max_value) = if stat.has_min_max_set() {
+        match stat {
+            ParquetStatistics::Boolean(s) => (
+                Some(ScalarValue::Boolean(Some(*s.min()))),
```

Comment:
This may look like a regression in performance, but I think it will actually perform better (as the old code is creating a single row array just to call an accumulator method)